Jim-Hodapp-Coaching/refactor-platform-rs
# Add OpenAPI documentation for the platform endpoints using utoipa

## Description
This PR adds [OpenAPI](https://swagger.io/docs/specification/about/) documentation for the platform endpoints using the crate [utoipa](https://docs.rs/utoipa/latest/utoipa/index.html).


#### GitHub Issue: N/A

### Changes
* Adds OpenAPI documentation of all existing endpoints for the organization_controller and user_session_controller
* Moves the controllers out of empty structs and just as modules, otherwise they're unusable types for utoipa's ToSchema derive macro
* Exposes a web-based RAPIDoc UI to view the API documentation and try the API out

### Testing Strategy
1. `cargo run  -- -l TRACE -d postgres://refactor:password@localhost:5432/refactor_platform`
2. Point your web browser to http://localhost:4000/rapidoc
3. You'll see something like:

<img width="1569" alt="Screenshot 2024-03-24 at 18 47 14" src="https://github.com/Jim-Hodapp-Coaching/refactor-platform-rs/assets/3219120/830ef76a-6600-42b6-8a7e-e7bf0b365316">

3. Try logging in with the `/login` endpoint
4. Every other endpoint should be functional then...the set-cookie session cookie gets set and forwarded by the OpenAPI doc UI
5. Try logging out with the `/logout` endpoint
6. Every endpoint should fail then until you log in again

### Concerns
None
